### PR TITLE
fix(guard): preserve browser MCP review context

### PR DESCRIPTION
**Pull request by kantorcodes**
## Summary

- preserve normalized browser intent on queued MCP approval requests
- classify in-memory screenshots as browser inspection instead of file upload
- attach the last confirmed selected or navigated page origin to current-page reviews
- remove URL fragments and redact sensitive query values before retaining page context

## Security boundaries

- timed grants remain limited to routine browser navigation, inspection, and interaction
- file transfer, privileged browser access, sensitive surfaces, and command execution remain excluded
- page context updates only after a successful MCP response

## Verification

- focused browser intent, temporary approval, approval queue, and MCP proxy tests
- Ruff check and format
- basedpyright: 0 errors
- installed-wheel classification check

## Summary by CodeRabbit

* **New Features**
  * Queued browser tool approvals now retain the associated browser intent.
  * Browser-intent tool calls now track and attach the current page URL for richer context.
* **Bug Fixes**
  * Improved routine browser action classification to reduce false filesystem-access signals.
  * Enhanced browser target URL extraction and redaction, including sensitive query values and fragment handling.
* **Tests**
  * Expanded browser MCP intent/redaction coverage, including screenshot-derived target fields and schema edge cases.
* **Chores**
  * Updated CI test-suite baseline and fixture checksum/diff report.

### CODE REVIEW

File: src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py [hunk 1]
```diff
@@ -324,7 +324,7 @@
 _SENSITIVE_AUTH_PATTERNS: tuple[str, ...] = ("auth_header", "authorization", "auth_token", "authtoken")
 _SENSITIVE_CDP_PATTERNS: tuple[str, ...] = ("cdp", "chrome_devtools_protocol", "raw_cdp")
 _SENSITIVE_SCRIPT_EVAL_PATTERNS: tuple[str, ...] = ("eval", "script", "javascript", "expression")
-_SENSITIVE_UPLOAD_PATTERNS: tuple[str, ...] = ("upload", "file_path", "filepath", "file_input")
+_SENSITIVE_UPLOAD_PATTERNS: tuple[str, ...] = ("upload", "file_input")
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py]: Fixed in 4f233dda0. Active file_path/filepath arguments again flag upload risk, while absent optional schema fields no longer misclassify in-memory screenshots.

File: src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py [hunk 2]
```diff
@@ -803,9 +806,9 @@ def _detect_sensitive_surfaces(
         surfaces.append("cdp")
     if any(p in all_text for p in _SENSITIVE_SCRIPT_EVAL_PATTERNS):
         surfaces.append("script_eval")
-    if any(p in all_text for p in _SENSITIVE_UPLOAD_PATTERNS):
+    if any(p in active_text for p in _SENSITIVE_UPLOAD_PATTERNS):
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py:822]: The schema-only behavior is intentional and remains fail-closed at the capability boundary: transfer operations stay browser.transfer and cannot receive reusable routine grants, while supplied file arguments activate filesystem and upload risk. Treating an absent optional screenshot filePath as an active upload caused the reported false positive.

File: src/codex_plugin_scanner/guard/proxy/runtime_mcp.py [hunk 3]
```diff
@@ -88,6 +94,42 @@ def _guard_action(value: object) -> GuardAction:
...
+)
+
+
+def _browser_url_from_arguments(arguments: object) -> str | None:
+    if not isinstance(arguments, dict):
+        return None
+    return _redacted_target_url(_extract_target_url(arguments))
+
+
+def _selected_browser_page_url(response: Mapping[str, object]) -> str | None:
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/proxy/runtime_mcp.py:102]: Fixed in 4f233dda0. Selected-page extraction is now restricted to MCP result.content text items; unrelated response fields cannot poison page context.

File: src/codex_plugin_scanner/guard/proxy/runtime_mcp.py [hunk 4]
```diff
@@ -2977,6 +3033,34 @@ def _allow_and_forward(
...
+            transport=self.transport,
+            server_identity=self._session_server_identity(),
+        )
+        if normalize_browser_mcp_intent(artifact, arguments) is None:
+            return
+        selected_url = _selected_browser_page_url(response)
+        if selected_url is not None:
+            self._browser_current_page_url = selected_url
+            return
+        if tool_name.lower() not in {"navigate_page", "new_page", "browser_navigate", "browser_new_page"}:
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/proxy/runtime_mcp.py]: Fixed in 4f233dda0. Context invalidation now derives from normalized browser intent rather than a hardcoded tool-name list.

File: src/codex_plugin_scanner/guard/proxy/runtime_mcp.py [hunk 5]
```diff
@@ -584,6 +626,7 @@ def __init__(
         self._active_executable_identity: dict[str, object] | None = None
         self._active_server_env_values_hash: str | None = None
         self._active_server_identity: McpServerIdentity | None = None
+        self._browser_current_page_url: str | None = None
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/proxy/runtime_mcp.py:615]: Fixed in 4f233dda0. Starting a new child process clears cached browser page context, with a regression test covering session reset.

File: src/codex_plugin_scanner/guard/proxy/runtime_mcp.py [hunk 6]
```diff
@@ -2977,6 +3033,34 @@ def _allow_and_forward(
...
+        )
+        if normalize_browser_mcp_intent(artifact, arguments) is None:
+            return
+        selected_url = _selected_browser_page_url(response)
+        if selected_url is not None:
+            self._browser_current_page_url = selected_url
+            return
+        if tool_name.lower() not in {"navigate_page", "new_page", "browser_navigate", "browser_new_page"}:
+            return
+        self._browser_current_page_url = _browser_url_from_arguments(arguments)
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/proxy/runtime_mcp.py]: Fixed in 580f23e6e. Any successful non-inspection browser action without a trusted selected-page response clears cached context, including navigation, interaction, transfer, and privileged/script actions.

File: src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py [hunk 7]
```diff
@@ -478,6 +478,8 @@ def normalize_browser_mcp_intent(
         return None
 
     target_url = _extract_target_url(mapping) if mapping else None
+    if target_url is None:
+        target_url = _optional_str(artifact.metadata.get("browser_current_page_url"))
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py:482]: Fixed in 4f233dda0. Requested navigation URLs are never cached as confirmed page context; only trusted selected-page response data can populate it, otherwise page-changing actions clear the cache.

File: src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py [hunk 8]
```diff
@@ -695,7 +697,7 @@ def _redacted_target_url(url: str | None) -> str | None:
         return url
 
     if not parsed.query:
-        return url
+        return urlunparse(parsed._replace(fragment=""))
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py]: Fixed in 4f233dda0. Safe hash-router fragments are preserved, sensitive key/value fragments are retained with values redacted, and opaque fragments are removed.

File: src/codex_plugin_scanner/guard/proxy/runtime_mcp.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/proxy/runtime_mcp.py:615]: Fixed in 4f233dda0. Child-process session reset now clears cached browser page context and the regression test verifies no cross-session carryover.

File: src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py [hunk 9]
```diff
@@ -686,30 +688,38 @@ def _normalize_path_prefix(url: str | None) -> str | None:
...
+
+
+def _redacted_url_pairs(pairs: list[tuple[str, str]]) -> list[tuple[str, str]]:
+    return [(key, "[redacted]" if _normalize_query_key(key) in _REDACT_QUERY_KEYS else value) for key, value in pairs]
+
+
+def _redacted_url_fragment(fragment: str) -> str:
+    if not fragment:
+        return ""
+    if fragment.startswith(("/", "!/")):
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply src/codex_plugin_scanner/guard/runtime/browser_mcp_intent.py:715]: Fixed in 4ed2942c5. Hash-router paths remain intact, while fragment query parameters now use the same sensitive-value redaction as ordinary queries; a regression covers token redaction with a preserved route and non-sensitive tab.
